### PR TITLE
Prepare data outside of the loop

### PR DESCRIPTION
**Pull request by ssheikin**

### CODE REVIEW

**Review by sopel39 — APPROVED** [review]

**Review by kokosing — APPROVED** [review]
Please ping me once you get a green light.

File: core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java [hunk 1]
```diff
@@ -355,10 +355,10 @@ default void dropColumn(ConnectorSession session, ConnectorTableHandle tableHand
     default Optional<ConnectorNewTableLayout> getInsertLayout(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         ConnectorTableProperties properties = getTableProperties(session, tableHandle);
+        Map<ColumnHandle, String> columnNamesByHandle = getColumnHandles(session, tableHandle).entrySet().stream()
+                .collect(toMap(Map.Entry::getValue, Map.Entry::getKey));
         return properties.getTablePartitioning()
                 .map(partitioning -> {
-                    Map<ColumnHandle, String> columnNamesByHandle = getColumnHandles(session, tableHandle).entrySet().stream()
```
Comment (findepi) [inline core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java:360]: it';s not a loop

**Review by findepi — CHANGES_REQUESTED** [review]

File: core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java (hunk 1 above)
Comment (ssheikin) [reply core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java:360]: @findepi could you please advice better message?

Comment (findepi) [reply core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java:360]: Unless I am wrong here, we're inside `Optional#map` call. 
Thus, extracting the call before makes the `getColumnHandles` being called unconditionally, also when not needed.

Comment (ssheikin) [reply core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java:360]: @findepi, thank you!

**Pull request closed without merging**